libobs/media-io/video-io.c

```diff
@@ -610,10 +667,30 @@ void video_output_dec_texture_encoders(video_t *video)
 
 void video_output_inc_texture_frames(video_t *video)
 {
-	os_atomic_inc_long(&video->total_frames);
+	os_atomic_inc_long(&get_root(video)->total_frames);
 }
 
 void video_output_inc_texture_skipped_frames(video_t *video)
 {
-	os_atomic_inc_long(&video->skipped_frames);
+	os_atomic_inc_long(&get_root(video)->skipped_frames);
+}
+
+video_t *video_output_create_with_frame_rate_divisor(video_t *video,
+						     uint32_t divisor)
+{
+	if (!video || divisor == 0)
```

Comment:
I'm not sure it's a good idea to allow creating a dedicated `video_t` instance if the divisor is `1`. May encourage bad behavior if it's plugin-facing.

Other question is do we have to worry about deduplication of video outputs here, or might it be deduplicated elsewhere?

Reply:
Creating instances with `divisor` set to `1` isn't super useful, agreed; I can add this to the check

As far as deduplication, making parent contexts keep track of derives contexts should be straightforward enough; though it'd probably also require some form of refcounting or similar for those contexts, as opposed to the clear ownership semantics that are being used right now

Are you worried about lots of video outputs being created with the same parent and divisor using up additional memory (and creating additional unused(?) threads), or do you have some other use case in mind for deduplicating?

Reply:
Added to https://github.com/obsproject/obs-studio/pull/9001/commits/f2d41e4627a416e55a5cba3cd3b0e214a2897f80 to address `divisor == 1` case